tests/e2e/workloads/test_data_consistency.py

```diff
@@ -108,6 +108,12 @@ def test_log_reader_writer_parallel(project, tmp_path):
     # is a bit larger than usual number required to reproduce bug from
     # BZ 1989301, but we need to be sure here)
     number_of_fetches = 120
+    # if given fetch fail, we will ignore the failure unless the number of
+    # failures is too high (this has no direct impact of feature under test,
+    # we should be able to detect the bug even with 10% of rsync failures,
+    # since data corruption doesn't simply go away ...)
+    number_of_failures = 0
+    allowed_failures = 12
```

Comment:
won't that increase the run time significantly?

Reply:
The number of iterations as specified in `number_of_fetches = 120` stays the same. And out of these 120 iterations, we allow 10% to recover from restarting container. The container restart we are trying to overcome should not take long time, so I don't expect the runtime to increase in a significant way.

That said, we can observe the number of failures (we are logging this info into debug logs), and test runtime in a full RC test matrix run, and see if that really holds, and tweak the limits accordingly.

Reply:
How was value of 12 achieved?

Reply:
> How was value of 12 determined?

It's 10% of 120 iterations. The value is a qualified guess based on observations of previous runs:
 
- I saw that when these container restarts happened, I noticed about 4 such restarts during one hour. The test itself takes about 5-10 minutes, so we could expect about 1 or 2 (at max) such failures during a single run (that said, this was based on a single observation).
- The problem caused fail of about 10% of test cases executed for a particular build.

So on one hand, using just 2 allowed failures could work, but there is a good chance that it won't be enough. So instead I decided to start with value from the upper range, to be sure that the test case is able to recover from the problem and get the actual data. Based on that, we can tweak the limits later (after at least on full test matrix run on a x.y release).

This should not cause any troubles wrt the ability of the test to detect regressions or new problems.